lib/utils.pm

```diff
@@ -1658,6 +1659,44 @@ sub script_output_retry {
     die("Waiting for Godot: $cmd") if $die;
 }
 
+
+=head2 validate_script_output_retry
+
+ validate_script_output_retry($cmd, $check, [retry => $retry], [delay => $delay], [timeout => $timeout]);
+
+Repeat command until validate_script_output succeeds or die. Return the output of the command on success.
+
+C<$retry> refers to the number of retries and defaults to C<10>.
+
+C<$delay> is the time between retries and defaults to C<30>.
+
+If the command doesn't succeed after C<$retry> retries,
+this function will die.
+
+Example:
+
+ validate_script_output_retry('ping -c1 -W1 machine', m/1 packets transmitted/, retry => 5, delay => 60);
+
+=cut
+sub validate_script_output_retry {
+    my ($cmd, $check, %args) = @_;
+    my $retry = $args{retry} // 10;
+    my $delay = $args{delay} // 30;
+    my $timeout = $args{timeout} // 90;
+
+    for (1 .. $retry) {
+        eval { validate_script_output($cmd, $check, timeout => $timeout); };
```

Comment:
the function `validate_script_output()` allow to pass arguments to the `script_output()` call, e.g. `proceed_on_failure => 1`

```suggestion
    my ($cmd, $check, %args) = @_;
    my %args = compat_args(
        {
            retry => 10,
            delay => 30,
            timeout => 90
        }, [], @_);

    my $retry = delete $args{retry};
    my $delay = delete $args{delay};

    for (1 .. $retry) {
        eval { validate_script_output($cmd, $check, %args); };
```